external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSchemaHelperSuite.scala

```diff
@@ -133,4 +133,27 @@ class AvroSchemaHelperSuite extends SQLTestUtils with SharedSparkSession {
       helperNullable.validateNoExtraCatalystFields(ignoreNullable = false)
     }
   }
+
+  test("validateNoExtraRequiredAvroFields detects required fields and ignores nullable fields") {
```

Comment:
Shall we aadd a JIRA prefix in the test title?

Reply:
yes, good call